examples/official-site/sqlpage/migrations/23_uploaded_file_functions.sql

```diff
@@ -105,6 +105,57 @@ select ''redirect'' as component,
 where sqlpage.uploaded_file_mime_type(''myfile'') not in (select mime_type from allowed_mime_types);
 ```
 '
+),
+(
+    'uploaded_file_name',
+    '0.23.0',
+    'file-description',
+    'Returns the `filename` value in the `content-disposition` header.
+
+## Example: saving uploaded file metadata for later download
+
+### Making a form
+
+```sql
+select ''form'' as component, ''handle_file_upload.sql'' as action;
+select ''myfile'' as name, ''file'' as type, ''File'' as label;
+```
+
+### Handling the form response
+
+### Inserting an arbitrary file as a [data URL](https://en.wikipedia.org/wiki/Data_URI_scheme) into the database
+
+In `handle_file_upload.sql`, one can process the form results like this:
+
+```sql
+insert into uploaded_files (fname, content, uploaded) values (
+  sqlpage.uploaded_file_name(''myfile''),
+  sqlpage.read_file_as_data_url(sqlpage.uploaded_file_path(''myfile'')),
+  CURRENT_TIMESTAMP
+);
+```
+
+> *Note*: Data URLs are larger than the original file, so it is not recommended to use them for large files.
+
+### Downloading the uploaded files
+
+The file can be downloaded by clicking a link like this:
+```sql
+select 'text' as component, (
+  select '<a download="'||fname||'" href="'||content||'">Click to download '||fname||'</a>' from uploaded_files
+  where fname = $file_name
+  limit 1
+) as html;
+```
```

Comment:
Unfortunately, that doesn't download the file. I tested with an image, and the browser just displays it. Also there is no way for the browser to know the file name. Apparently, there has been some support for including the file name as a media parameter in the data uri, but seems the "download" attribute is the most supported way.

I guess we need some handlebars to make a download component...

Reply:
You are right, if the file can be displayed in the browser directly, it will (and can be saved to disk from there). 

We should add a new `download` row-level attribute to the button component. But we can do that in a separate pull request. 

For the moment, can you apply the change I suggest above to the example? You can change it to add the download attribute in the next pr.